packages/monitoring-system/data-processor/src/data-processors/cloud-tasks-data-processor.ts

```diff
@@ -13,9 +13,127 @@
 // limitations under the License.
 //
 import {DataProcessor} from './data-processor-abstract';
+import {Firestore, WriteResult} from '@google-cloud/firestore';
+import {CloudTasksClient, protos, v2} from '@google-cloud/tasks';
 
+type CloudTasksList = [
+  protos.google.cloud.tasks.v2.ITask[],
+  protos.google.cloud.tasks.v2.IListTasksRequest | null,
+  protos.google.cloud.tasks.v2.IListTasksResponse
+];
+
+interface QueueStatus {
+  [queueName: string]: number;
+}
+
+/**
+ * Collects the current status of Cloud Task queues associated
+ * with repo automation bots and inserts it into Firestore
+ */
 export class CloudTasksProcessor extends DataProcessor {
+  private PROJECT_ID = 'repo-automation-bots';
+  private LOCATION = 'us-central1';
```

Comment:
Not sure if `PROJECT_ID` and `LOCATION` should be hardcoded (either in the class like here, or in a separate config file) or if we should be pulling this information from GCP somehow. There's no guarantee though that this data processor will always be in the same project/location as the bots so we can't just use the 'current' project/location.